cli/EAP7-979_Allow_CLI_output_scrolling.adoc

```diff
@@ -0,0 +1,117 @@
+= EAP7-979 WFCORE-3568 Allow CLI output scrolling
+:author:            Marek Marusic
+:email:             mmarusic@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          cli,output,scrolling
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Since https://issues.jboss.org/browse/WFCORE-426[WFCORE-426] and https://issues.jboss.org/browse/EAP7-735[EAP7-735]
+mouse scrolling is forbidden to large CLI out and user has to press Down or Enter to navigate.
+After pressing Down or Enter to display more output, user can not go up by pressing Up.
+Therefore, it would be convenient to allow CLI output scrolling by mouse and by Up/Down and PgUp/PgDown keys.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-979[EAP7-979]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFCORE-3568[WFCORE-3568]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+* mailto:eduda@redhat.com[Erich Duda]
+
+=== Affected Projects or Components
+
+* CLI
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+* Fast scrolling using PgUp / PgDn. (ATM there is possibility to use space for fast scroll down)
+* Scrolling by mouse
+* Ability to scroll up
+* Since as mentioned in https://developer.jboss.org/wiki/CLI-PagingLongCLIOutput#comment-16541
+  the "less" behaviour would result in buffer alternation and therefore after
+  exiting the reading of output the output wouldn't be visible any more in the main buffer of the CLI.
+  This is not convenient since users may want to see at least part of the output
+  in the main buffer.
+  Therefore I would propose to print a part of the output from start to the last scrolled position
+  into the main buffer of the CLI
+  when exiting the more/less reading.
+
+
+==== List of keys and actions:
+Composed keys are unstable during the testing
+(this is tracked in https://issues.jboss.org/browse/WFCORE-3545[WFCORE-3545])
+Therefore I propose to use for every action a noncomposed key as well.
+This will enable us to create a stable tests.
+
+* PGDOWN / SPACE : scroll one page down
+* PGUP / BACKSLASH : scroll one page up
+* UP / SEMI_COLON: scroll one line up
+* DOWN / ENTER / CTRL_M : scroll one line down
+* Mouse scroll up : scroll up
+* Mouse scroll down : scroll down
+* ESC / 'q' / CTRL_C : quit
+* HOME : go to start of the document
+* END : go to end of the document
+
+=== Nice-to-Have Requirements
+
+Support for searching of patterns. Similar as it is done by less command.
```

Comment:
I have done an analysis for the search functionality as requested in the review.

However, the original issue [2] did not mention a possibility to search patterns and I think it would require a bigger changes.
Therefore, it could be better to create a separate "less" command for the search part and track that in separate Feature Request / Issue. The "less" implemented as a command would give user the possibility to choose whether he wants to use the auto "more" functionality or whether he wants to use the "less" command.
The "less" command could be used for operations like ":read-resource()" through a pipe e.g. ":read-resource() | less".

@jfdenise @dudaerich  what do you think about this ?